client/packages/reports/src/DetailView/DetailView.tsx

```diff
@@ -180,14 +189,24 @@ const DetailViewInner = ({
       <ReportArgumentsModal
         key={report.id}
         report={reportWithArgs}
-        onReset={() => setReportWithArgs(undefined)}
+        onReset={() => {
+          setReportWithArgs(undefined);
+          setReportLoading(true);
+        }}
         onArgumentsSelected={generateReport}
       />
-
-      {!fileId ? (
-        <NothingHere body={errorMessage} />
+      {reportLoading && (
+        <BasicSpinner messageKey="messages.loading-report"></BasicSpinner>
+      )}
+      {fileId ? (
+        <iframe
+          src={url}
+          width="100%"
+          style={{ borderWidth: 0 }}
+          onLoad={loaded}
```

Comment:
```suggestion
          onLoad={() => {
            setReportLoading(false);
          }}
```
The `loaded` function was a bit confusing to me, feels like this would be clearer? But I'm ok with how it is if you disagree.

Reply:
Agree! I like your suggestion better also